server/src/testFixtures/java/io/crate/testing/SQLExecutor.java

```diff
@@ -720,47 +723,89 @@ public SQLExecutor addPartitionedTable(String createTableStmt, Settings customSe
-        IndexTemplateMetadata.Builder template = IndexTemplateMetadata.builder(boundCreateTable.templateName())
-            .patterns(singletonList(boundCreateTable.templatePrefix()))
-            .putMapping(new CompressedXContent(BytesReference.bytes(mappingBuilder)))
-            .settings(buildSettings(false, combinedSettings, prevState.nodes().getSmallestNonClientNodeVersion()))
-            .putAlias(alias);
 
-        mdBuilder.put(template);
+        Version smallestVersion = prevState.nodes().getSmallestNonClientNodeVersion();
+        Version versionCreated = IndexMetadata.SETTING_INDEX_VERSION_CREATED.get(combinedSettings);
 
-        RoutingTable.Builder routingBuilder = RoutingTable.builder(prevState.routingTable());
+        // skip mapping generation if using RelationMetadata
+        // This is a bit different than production code but is done to avoids issues with duplicate oid generation
+        // (which would happen once here, and once again further done via mdBuilder.setTable)
+        Map<String, Object> mapping = versionCreated.before(Version.V_6_0_0)
+            ? mapping = TestingHelpers.toMapping(columnOidSupplier, boundCreateTable)
+            : (partitioned ? Map.of() : null);
+        AliasMetadata alias = new AliasMetadata(boundCreateTable.tableName().indexNameOrAlias());
+        if (partitioned) {
+            XContentBuilder mappingBuilder =
+                JsonXContent.builder().map(Map.of(Constants.DEFAULT_MAPPING_TYPE, mapping));
+            IndexTemplateMetadata.Builder template = IndexTemplateMetadata.builder(boundCreateTable.templateName())
+                .patterns(singletonList(boundCreateTable.templatePrefix()))
+                .settings(buildSettings(false, combinedSettings, smallestVersion))
+                .putMapping(new CompressedXContent(BytesReference.bytes(mappingBuilder)))
+                .putAlias(alias);
+            mdBuilder.put(template);
+        }
+        List<String> indexUUIDs = new ArrayList<>();
+        ClusterBlocks.Builder blocksBuilder = ClusterBlocks.builder()
+            .blocks(prevState.blocks());
         for (String partition : partitions) {
-            IndexMetadata indexMetadata = getIndexMetadata(
+            IndexMetadata.Builder imdBuilder = getIndexMetadata(
                 partition,
                 combinedSettings,
                 mapping, // Each partition has the same mapping.
-                prevState.nodes().getSmallestNonClientNodeVersion())
-                .putAlias(alias)
-                .build();
+                smallestVersion
+            );
+            if (partitioned) {
+                imdBuilder.putAlias(alias);
+            }
+            IndexMetadata indexMetadata = imdBuilder.build();
             mdBuilder.put(indexMetadata, true);
             routingBuilder.addAsNew(indexMetadata);
+            indexUUIDs.add(indexMetadata.getIndexUUID());
+            blocksBuilder.addBlocks(indexMetadata);
+        }
```

Comment:
Shouldn't this only be applied for `versionCrearted.before(Version.V_6_0_0)`?

Reply:
Not sure which part exactly you mean? We're still going to have IndexMetadata per partition with 6.0.0 and later.

Reply:
Ah sorry, I had the assumption that `setTable` will also add the required IndexMetadata, but of course doesn't make sense.